### ISSUE TITLE
PHP 8.4: Implicitly nullable parameter declarations deprecated

### ISSUE THREAD

**Issue by antoniovj1** [widmogrod/php-monads#113]
Hi!

I found a deprecation with PHP 8.4.

https://php.watch/versions/8.4/implicitly-marking-parameter-type-nullable-deprecated

```
!!  Deprecated: Widmogrod\Functional\map(): Implicitly marking parameter $value as nullable is deprecated, the explicit nullable type must be used instead in /var/www/html/vendor/widmogrod/php-functional/src/Functional/functions.php on line 171
!!  PHP Deprecated:  Widmogrod\Functional\map(): Implicitly marking parameter $value as nullable is deprecated, the explicit nullable type must be used instead in /var/www/html/vendor/widmogrod/php-functional/src/Functional/functions.php on line 171
!!  PHP Deprecated:  Widmogrod\Functional\bind(): Implicitly marking parameter $value as nullable is deprecated, the explicit nullable type must be used instead in /var/www/html/vendor/widmogrod/php-functional/src/Functional/functions.php on line 191
```

This apply to multiple functions

**Comment by mihaileu** [comment]
@widmogrod can you take a look please

**Comment by widmogrod** [comment]
Working on it https://github.com/widmogrod/php-functional/pull/115

**Comment by widmogrod** [comment]
New tag released `7.0.0`

**Comment by antoniovj1** [comment]
Thank you very much!
